test/unit/gateways/braintree_blue_test.rb

```diff
@@ -932,6 +932,13 @@ def test_solution_id_is_added_to_create_transaction_parameters
     ActiveMerchant::Billing::BraintreeBlueGateway.application_id = nil
   end
 
+  def test_override_application_id_is_sent_to_channel
+    gateway = BraintreeBlueGateway.new(merchant_id: 'test', public_key: 'test', private_key: 'test', channel: 'overidden-channel')
+    assert_equal gateway.send(:create_transaction_parameters, 100, credit_card('41111111111111111111'), {})[:channel], 'overidden-channel'
+
+    assert_equal gateway.send(:create_transaction_parameters, 100, credit_card('41111111111111111111'), { override_application_id: 'override-application-id' })[:channel], 'override-application-id'
```

Comment:
Non blocking suggestion: sweeten this test a little bit. The addition of the options hash seems to push it into overstuffed-one-liner territory.

I'm talking as simple as
```
    gateway_response = gateway.send(:create_transaction_parameters, 100, credit_card('41111111111111111111'), { override_application_id: 'override-application-id' })
    assert_equal gateway_response[:channel], 'override-application-id'
```

... and then change up the first one to mirror that pattern. The first one, on its own, I wouldn't quite say seems overstuffed, but no point in spelling them different.